src/app/utils/performance-tests.spec.ts

```diff
@@ -0,0 +1,123 @@
+/*
+ *     This file is part of ndb-core.
+ *
+ *     ndb-core is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     ndb-core is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { async, TestBed } from "@angular/core/testing";
+import { SessionService } from "../core/session/session-service/session.service";
+import { AppConfig } from "../core/app-config/app-config";
+import { AppModule } from "../app.module";
+import { SyncState } from "../core/session/session-states/sync-state.enum";
+import moment from "moment";
+import { ChildrenService } from "../child-dev-project/children/children.service";
+import { deleteDB } from "idb";
+
+/**
+ * These performance tests are actually integration tests that interact with a remote database.
+ *
+ * You need to enable CORS for the tests to run by editing karma.conf.js replacing `browsers: ['Chrome'],` with the following:
+browsers: ['Chrome_without_security'],
+customLaunchers:{
+  Chrome_without_security:{
+    base: 'Chrome',
+    flags: ['--disable-web-security']
+  }
+},
+ */
+xdescribe("Performance Tests", () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [AppModule],
+    }).compileComponents();
+
+    AppConfig.settings = {
+      database: {
+        name: "app",
+        remote_url: "http://dev.aam-digital.com/db/",
+        timeout: 60000,
+        useTemporaryDatabase: false,
+      },
+    } as any;
+
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 150000;
+  }));
+
+  it("sync initial and indexing", async () => {
+    // delete previously synced database; uncomment this to start with a clean state and test an initial sync.
+    // await deleteAllIndexedDB(db => true);
+
+    const session = TestBed.inject<SessionService>(SessionService);
+    const syncTimer = new Timer(true);
+    await session.login("demo", "pass");
```

Comment:
is it okay to have credentials as plaintext here?

Reply:
good point. even though the credentials are trivial this shouldn't be in the code base

Reply:
a really clean solution would probably be an .env file to set environment variables and put that file on .gitignore. That way we could even easily configure these in the CI as environment variables.

But for now, I am taking the shortcut on just putting variables with placeholders to finally get this PR out of the door.